pawnShop/src/main/java/com/example/pawnShop/Controller/ProductTypeController.java

```diff
@@ -0,0 +1,48 @@
+package com.example.pawnShop.Controller;
+
+import com.example.pawnShop.Dto.Product.ProductTypeDto;
+import com.example.pawnShop.Service.Contract.ProductTypeService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/product_type")
+public class ProductTypeController {
+    private final ProductTypeService productTypeService;
+
+    @GetMapping("/get_all")
+    public ResponseEntity<List<ProductTypeDto>> getAllProductsTypes(){
+        var productsTypes = productTypeService.getAllProductsTypes();
+
+        return ResponseEntity.ok(productsTypes);
+    }
+    @GetMapping("/get_by_id")
+    public ResponseEntity<ProductTypeDto> getProductTypeById(@RequestParam UUID id){
+        var productType = productTypeService.getProductTypeById(id);
+
+        return ResponseEntity.ok(productType);
+    }
+    @PostMapping("/add")
+    public ResponseEntity<String> addProductType(@RequestParam String productTypeName){
+        var newProductTypeName = productTypeService.addProductType(productTypeName);
+
+        return ResponseEntity.ok(newProductTypeName);
+    }
+    @PostMapping("/modify")
+    public ResponseEntity<String> addProductType(@RequestBody ProductTypeDto modifiedProductType){
+        var modifiedProductTypeName = productTypeService.updateProductType(modifiedProductType);
```

Comment:
typically it's good to use Optional class:
Optional<ProductTypeDTO> modifiedProductTypeName
https://medium.com/javarevisited/optional-class-in-java-8-making-your-code-more-clear-and-concise-62af0712910d
what does this mean? 
Imagine if the UI sends an object to the backend, but to object's value is Null. What will happen then ? 
An exception would be thrown , because you can not update/add/save etc. nullable objects. Using Optional.empty() afterwards would save you from this scenario ;)